pkg/envoy/ads/response.go

```diff
@@ -42,31 +35,12 @@ func (s *Server) sendTypeResponse(typeURI envoy.TypeURI, proxy *envoy.Proxy, ser
 // for, and will have them sent to the proxy server.
 // If no DiscoveryRequest is passed, an empty one for the TypeURI is created
 // TODO(draychev): Convert to variadic function: https://github.com/openservicemesh/osm/issues/3127
-func (s *Server) sendResponse(typeURIsToSend mapset.Set,
-	proxy *envoy.Proxy,
-	server *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer,
-	request *xds_discovery.DiscoveryRequest,
-	cfg configurator.Configurator) error {
-	// Order is important: CDS, EDS, LDS, RDS
-	// See: https://github.com/envoyproxy/go-control-plane/issues/59
-	var typesToSend []envoy.TypeURI
-	for _, typeURI := range envoy.XDSResponseOrder {
-		if typeURIsToSend.Contains(typeURI) {
-			typesToSend = append(typesToSend, typeURI)
-		}
-	}
-
-	if len(typesToSend) < 1 {
-		// This should never happen.
-		log.Error().Msg("The list of discovery responses passed to sendResponse() is empty. This is an invalid invocation of sendResponse().")
-		return errors.New("list of discovery responses is empty")
-	}
-
+func (s *Server) sendResponse(proxy *envoy.Proxy, server *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer, request *xds_discovery.DiscoveryRequest, cfg configurator.Configurator, typeURIsToSend ...envoy.TypeURI) error {
 	thereWereErrors := false
 
 	// Order is important: CDS, EDS, LDS, RDS
 	// See: https://github.com/envoyproxy/go-control-plane/issues/59
-	for _, typeURI := range typesToSend {
+	for _, typeURI := range typeURIsToSend {
```

Comment:
@eduser25 looking at the `newJob()` invocations - it seems to me that the only use cases of the list of TypeURIs is with these:
 - `envoy.TypeWildcard`
 - `envoy.XDSResponseOrder`
 - `[]envoy.TypeURI{envoy.TypeSDS}`

So I think we are removing very complex piece of code and documenting the risk  with the few mentions of "order is important".

Reply:
>  it seems to me that the only use cases of the list of TypeURIs is with these

I'll bet this will change in 3-6 months, and it's likely new blood will use this functions. If you want to keep it as is I don't see a problem, but then I would suggest to add more wording on the function header to make sure all the intricacies of its use are documented.

To me it's still be easier to enforce proper use on the function signature than doing this, but up to you.